spot_driver/spot_driver/spot_ros2.py

```diff
@@ -2090,6 +2174,58 @@ def handle_navigate_to(self, goal_handle: ServerGoalHandle) -> NavigateTo.Result
 
         return result
 
+    def handle_navigate_to_dynamic_feedback(self) -> None:
+        """Thread function to send navigate_to_dynamic feedback"""
+        if self.spot_wrapper is None:
+            return
+
+        while rclpy.ok() and self.run_navigate_to_dynamic:
+            localization_state = self.spot_wrapper._graph_nav_client.get_localization_state()
+            if localization_state.localization.waypoint_id:
+                feedback = NavigateToDynamic.Feedback()
+                feedback.waypoint_id = localization_state.localization.waypoint_id
+                feedback.message = self.spot_wrapper._navigate_to_dynamic_feedback
+                if self.goal_handle_dynamic is not None:
+                    self.goal_handle_dynamic.publish_feedback(feedback)
+            time.sleep(1)
+
+    def handle_navigate_to_dynamic(self, goal_handle: ServerGoalHandle) -> NavigateToDynamic.Result:
+        """Navigate to a given goal waypoint. Unlike navigate_to,
+        this method supports velocity adjustment and pausing/cancellation during a navigation request
+        (see handle_set_navigate_to_params)"""
+
+        self.goal_handle_dynamic = goal_handle
+        # Create thread to periodically publish feedback
+        feedback_thread = threading.Thread(target=self.handle_navigate_to_dynamic_feedback, args=())
```

Comment:
Is there a way to do this without creating a separate thread for feedback? There are already some threading bottlenecks that we are working on improving